homeassistant/components/zeroconf/__init__.py

```diff
@@ -42,35 +41,33 @@
     info = ServiceInfo(ZEROCONF_TYPE, zeroconf_name,
                        port=hass.http.server_port, properties=params)
 
-    zeroconf = Zeroconf(hass.loop)
+    zeroconf = Zeroconf()
 
-    await zeroconf.register_service(info)
+    zeroconf.register_service(info)
 
-    async def new_service(service_type, name):
-        """Signal new service discovered."""
-        service_info = await zeroconf.get_service_info(service_type, name)
-        info = info_from_service(service_info)
-        _LOGGER.debug("Discovered new device %s %s", name, info)
-
-        for domain in zeroconf_manifest.SERVICE_TYPES[service_type]:
-            await hass.config_entries.flow.async_init(
-                domain, context={'source': DOMAIN}, data=info
-            )
-
-    def service_update(_, service_type, name, state_change):
+    def service_update(zeroconf, service_type, name, state_change):
         """Service state changed."""
         if state_change is ServiceStateChange.Added:
-            hass.async_create_task(new_service(service_type, name))
-
-    for service in zeroconf_manifest.SERVICE_TYPES:
+            service_info = zeroconf.get_service_info(service_type, name)
+            info = info_from_service(service_info)
+            _LOGGER.debug("Discovered new device %s %s", name, info)
+
+            for domain in ZEROCONF[service_type]:
+                hass.add_job(
```

Comment:
This is fine. Calling an async function creates a coroutine which is then scheduled.